### ISSUE TITLE
Ionization / FRESH / Air Magic mode

### ISSUE THREAD

**Issue by m4rkireland** [wuwentao/midea_ac_lan#175]
### Device type and model (or SN)

Air Conditioner 00000Q1B

### The description of new feature

Originally listed here https://github.com/georgezhao2010/midea_ac_lan/issues/67 and marked as "not planned" by the original maintainer, I'd like to re-request this feature please.  The original request by user @IGaMbLeRI was as follows:

> Some models of Midea (and other compatible brands) A/C and (de)humidifier units have option to switch on Ionization by pressing a button called "FRESH" on the remote control. For example, if you see [this manual](https://github.com/georgezhao2010/midea_ac_lan/files/9387140/13639883_msfaau-12hrfn8-qrd6gw.pdf) for what seems to be BreezeleSS+, on PDF Page 8, you will see how it indicates that the Ionization mode is working - leaf icon on the LED panel.
> 
> I personally have [a different A/C brand and model](https://www.climatecnica.gr/images/pdf/OMNIA_ECO_03MVI32_UM.pdf) but it also has that same "FRESH" feature that is present in many other Midea devices of different classes.
> 
> Is it possible to have this feature added for compatible devices with a toggle, just like "ECO Mode", "Prompt Tone", "Swing Vertical" etc?
> 
> Kind regards.

I have a Midea Xtreme Eco `Air Conditioner 00000Q1B` and the "Air Magic" function is a separate button on the remote control, to augment the currently selected mode, rather than a separate "mode" eg, Air Magic can be active when the overall mode is "Heat" or "Cool" or "Auto" or "Dry" or "Fan Only" etc.

I am willing to help in any way I can, wireshark, debug logs etc.

Thanks very much

**Comment by chemelli74** [comment]
if it's a separate button it can be represented with a switch in HA, isn't it ?

**Comment by m4rkireland** [comment]
Yes it could be added as a switch, however the issue is that the integration doesn't currently surface the 'Air Magic' function to HA, so there is nothing for the switch to control. As far as I am aware anyway!

**Comment by rokam** [comment]
Are you able to control this feature using the app?

**Comment by m4rkireland** [comment]
It doesn't appear like I can control it through the Media 'Smart Home' app. Maybe that is our answer 😭

**Comment by morph027** [comment]
Looks like I can control it via the app

![Screenshot_20240918_171709_SmartHome](https://github.com/user-attachments/assets/ecf8ac8a-2a43-4443-992a-512bdc1ea1a8)

**Comment by m4rkireland** [comment]
Can we reopen this please?

**Comment by morph027** [comment]
Looks like the feature is present in https://github.com/mill1000/midea-ac-py

Probably have a look at the implementation.

**Comment by M-K3y** [comment]
I can confirm this issue also affects my device.

Device: Midea PortaSplit (portable AC unit)  
Integration: Midea AC LAN v0.6.9

The unit supports an Ionizer (ION) mode via the original remote control, but the integration does not expose any switch entity to control ION from Home Assistant.  
Whenever I enable ION manually, any control action from HA turns it off again.

It would be great if ION could be added as an optional switch entity, similar to ECO or Swing mode, so it can be toggled and maintained via HA without being reset.

**Comment by apfelsheriff** [comment]
Could we bring this topic up again? 
For the Portasplit device, the ION feature is definitely visible and managable in the Smarthome App as a switch.
